DNN Platform/DotNetNuke.Web/DependencyInjection/BuildUpExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.DependencyInjection.Extensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Web.Http.Filters;
+
+    using DotNetNuke.Common.Utilities;
+
+    /// <summary>
+    /// Adds property injection extension methods.
+    /// </summary>
+    public static class BuildUpExtensions
+    {
+        /// <summary>
+        /// Injects property dependency for public or proetected
+        /// properties that are decorated with <see cref="DependencyAttribute"/>.
+        /// </summary>
+        /// <param name="container">The service provider.</param>
+        /// <param name="filter">The <see cref="IFilter"/> to inject properties.</param>
+        public static void BuildUp(this IServiceProvider container, IFilter filter)
+        {
+            if (container == null || filter == null)
+            {
+                return;
+            }
+
+            var properties = GetDependencyProperties(filter.GetType());
+            foreach (var property in properties)
+            {
+                var service = container.GetService(property.GetType());
+                if (service != null)
+                {
+                    property.SetValue(service, filter);
+                }
+            }
+
+            IEnumerable<PropertyInfo> GetDependencyProperties(Type type)
+            {
+                return CBO.Instance.GetCachedObject<IEnumerable<PropertyInfo>>(
+                    new CacheItemArgs($"WebApiDependencyProperties_{type.FullName}"),
+                    (args) => type
+                        .GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
+                        ?.Where(propertyInfo => propertyInfo.GetCustomAttribute<DependencyAttribute>() != null),
```

Comment:
I'm removing the `?.` from the `Where` call b/c [`GetProperties`](https://docs.microsoft.com/en-us/dotnet/api/system.type.getproperties?view=netframework-4.8#System_Type_GetProperties_System_Reflection_BindingFlags_) should never return `null`.

I'm also adding `ToList` b/c we want to cache the results of the query.  Otherwise we're just caching the query, and it'll go back to the data source each time it's called.

```suggestion
                        .Where(propertyInfo => propertyInfo.GetCustomAttribute<DependencyAttribute>() != null)
                        .ToList(),
```

Reply:
This makes sense, thank you for the explanation